Makefile

```diff
@@ -1,15 +1,30 @@
 K3DCLUSTERNAME := wasm-cluster
-K3DSHIMIMAGENAME := ghcr.io/deislabs/containerd-wasm-shims/examples/k3d:v0.9.0
+K3DSHIMIMAGENAME := ghcr.io/deislabs/containerd-wasm-shims/examples/k3d:v0.9.1
 DOCKERDIR := ./wasm-shims/deployments/k3d
-APPSDIR := ./apps
+APPSDIR := ./apps/mqtt
+PORTFORWARDING := -p '8001:30010@loadbalancer' -p '8002:80@loadbalancer' -p '1883:31883@loadbalancer'
+redis_communication: APPSDIR := ./apps/redis
+redis_communication: PORTFORWARDING := -p '8001:30010@loadbalancer' -p '8002:80@loadbalancer'
+APPSDIRRUNTIME := ./apps/runtime
 
-all: create_k3d_cluster install_redis deploy_app run_integrationtest
+all: mqtt_communication
+
+mqtt_communication: create_k3d_cluster upgrade_spin_shim install_redis install_mosquitto deploy_app run_integrationtest
+
+redis_communication: create_k3d_cluster install_redis deploy_app run_integrationtest
 
 create_k3d_cluster:
 	@echo "Creating k3d cluster..."
-	k3d cluster create $(K3DCLUSTERNAME) --image $(K3DSHIMIMAGENAME) --api-port 6551 -p '8001:30010@loadbalancer' -p '8002:80@loadbalancer' --servers 1
+	k3d cluster create $(K3DCLUSTERNAME) --image $(K3DSHIMIMAGENAME) --api-port 6551 $(PORTFORWARDING) --servers 1
 	@echo "Loading spin runtime..."
-	kubectl apply -f ./apps/runtime/runtime.yaml
+	kubectl apply -f $(APPSDIRRUNTIME)/runtime.yaml
+
+upgrade_spin_shim: create_k3d_cluster
+	@echo "Upgrading spin shim..."
+	k3d cluster stop $(K3DCLUSTERNAME)
+	docker cp -a $(APPSDIRRUNTIME)/containerd-shim-spin-v1 k3d-$(K3DCLUSTERNAME)-server-0:/bin/containerd-shim-spin-v1
+	k3d cluster start $(K3DCLUSTERNAME)
+	docker exec -it k3d-$(K3DCLUSTERNAME)-server-0 chmod -R 777 ./bin/containerd-shim-spin-v1
```

Comment:
changed to 500 because execute permission for user is needed and this is the minimum you can apply according to permission table:
https://www.multacom.com/faq/password_protection/file_permissions.htm#:~:text=777%20%2D%20all%20can%20read%2Fwrite,group%2Fothers%20can%20read%20only.